server/qrops.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	return keyUID[0]
+}
+
+func getImageName(params url.Values, allowProfileImages bool, logger *zap.Logger) string {
+	imageName, ok := params["imageName"]
+
+	// if allowProfileImage was also not passed
+	// in that case we don't care and can return an empty string back
+	if !allowProfileImages {
+		return ""
+	}
+
+	//if the imageName was not passed, we default to "thumbnail"
+	if !ok || len(imageName) == 0 {
+		return "thumbnail"
+	}
+
+	return imageName[0]
+}
+
+func getSize(params url.Values, logger *zap.Logger) int {
+	size, ok := params["size"]
+
+	//if the size was not passed, we return nil;
+	if !ok || len(size) == 0 {
+		return 0
+	}
+	imageToBeResized, err := strconv.Atoi(size[0])
+
+	if err != nil {
+		logger.Error("[qrops-error] could not convert string to int for size param ", zap.Error(err))
+		return 0
+	}
+
+	return imageToBeResized
+}
+
+func generateQRBytes(params url.Values, logger *zap.Logger, multiaccountsDB *multiaccounts.Database) []byte {
+
+	qrURLBase64Decoded := getQrURL(params, logger)
+	superImposeLogoOnQR := getAllowProfileImage(params, logger)
+	level := getLevel(params, logger)
+	correctionLevel := qrcode.ErrorCorrectionMedium
+	keyUID := getKeyUID(params, superImposeLogoOnQR, logger)
+	imageName := getImageName(params, superImposeLogoOnQR, logger)
+	size := getSize(params, logger)
+
+	switch level {
+	case "1":
+		correctionLevel = qrcode.ErrorCorrectionLow
+	case "2":
+		correctionLevel = qrcode.ErrorCorrectionMedium
+	case "3":
+		correctionLevel = qrcode.ErrorCorrectionQuart
+	case "4":
+		correctionLevel = qrcode.ErrorCorrectionHighest
+	}
+
+	buf := NewWriterCloserByteBuffer()
+
+	qrc, err := qrcode.NewWith(qrURLBase64Decoded,
+		qrcode.WithEncodingMode(qrcode.EncModeAuto),
+		qrcode.WithErrorCorrectionLevel(correctionLevel),
+	)
+
+	if err != nil {
+		logger.Error("could not generate QRCode with provided options", zap.Error(err))
+		return nil
+	}
+
+	nw := standard.NewWithWriter(buf)
+	err = qrc.Save(nw)
+
+	if err != nil {
+		logger.Error("could not save image", zap.Error(err))
+		return nil
+	}
+
+	payload := buf.Bytes()
+
+	if superImposeLogoOnQR {
+		logo, err := GetLogoImage(multiaccountsDB, keyUID, imageName)
+
+		if err != nil {
+			logger.Error("could not get logo image from multiaccountsDB", zap.Error(err))
+			return nil
+		}
+
+		qrWidth, qrHeight, err := images.GetImageDimensions(payload)
+
+		if err != nil {
+			logger.Error("could not get image dimensions from payload", zap.Error(err))
+			return nil
+		}
+
+		logo, err = images.ResizeImage(logo, qrWidth/5, qrHeight/5)
+
+		if err != nil {
+			logger.Error("could not resize logo image ", zap.Error(err))
+			return nil
+		}
+
+		payload = images.SuperimposeLogoOnQRImage(payload, logo)
+	}
+
+	if size > 0 {
+
+		if err != nil {
+			logger.Error("could not convert string to int for size param ", zap.Error(err))
+			return nil
+		}
```

Comment:
You seem to be evaluating an error that isn't created.

Reply:
oh lol, somehow that got left in by mistake, will fix :)

Reply:
fixed here : https://github.com/status-im/status-go/pull/3172/commits/b5b8339943e3287dea723e4ce2b4f95686c6f88f